shared/continuations/lib/RegisterBuffer.cpp

```diff
@@ -192,18 +192,15 @@ Value *RegisterBufferPass::computeMemAddr(IRBuilder<> &Builder,
   } else if (auto *Inst = dyn_cast<CastInst>(Address)) {
     auto *Src = Inst->getOperand(0);
     Value *MemSrc = computeMemAddr(Builder, Src);
-    New = Builder.CreateCast(
-        Inst->getOpcode(), MemSrc,
-        PointerType::getWithSamePointeeType(
-            cast<PointerType>(Inst->getDestTy()), Data.Addrspace));
+    New = Builder.CreateCast(Inst->getOpcode(), MemSrc,
+                             PointerType::get(Inst->getType(), Data.Addrspace));
```

Comment:
I think this is incorrect: `inst->getType()` is the pointer type, but `PointerType::get` expects the element type.

Reply:
I think it is not a case anymore with opaque pointers. `PointerType::get` is just taking `->getContext()` from the Type, but I can change this to `PointerType::get(Inst->getType()->getContext()` ?

Reply:
With opaque pointers, this is just returning `ptr addrspace(..)` type, so it is correct (because the element type does not matter and is ignored).

But this code needs to function with typed pointers for the time being, and in that case it is incorrect.